Sources/SwiftProtobuf/JSONEncodingVisitor.swift

```diff
@@ -277,10 +280,14 @@ internal struct JSONEncodingVisitor: Visitor {
   }
 
   mutating func visitRepeatedEnumField<E: Enum>(value: [E], fieldNumber: Int) throws {
+    let options = self.options
     let alwaysPrintEnumsAsInts = options.alwaysPrintEnumsAsInts
     try _visitRepeated(value: value, fieldNumber: fieldNumber) {
       (encoder: inout JSONEncoder, v: E) throws in
-      if !alwaysPrintEnumsAsInts, let n = v.name {
+      if let e = v as? _CustomJSONCodable {
```

Comment:
Do the class level check so you can consult a boolean inside the loop for the common case and do this check more quickly?

Reply:
That won't really help.  `v` still needs to be cast to `_CustomJSONCodable` to access the protocol method on the instance.  That cast requires an `as?` or `as!` operation and those are implemented the same way in the runtime.  That's why I plan to experiment with changing `encodedJSONString` to a static class method that accepts an instance, as that would allow us to do a single test outside the loop.

Reply:
You lost me.  You'd completely skip the cast if the class doesn't conform, isn't that a huge win for the common case?

Reply:
#1058 is what I was thinking.